src/app/components/client/Chart.tsx

```diff
@@ -230,3 +230,111 @@ export const DoughnutChart = (props: Props) => {
     </>
   );
 };
+
+export type WelcomeToPremiumProps = {
+  data: Array<[string, number]>;
+  exposureReduction: number;
+};
+export const WelcomeToPremiumDonutChart = (props: WelcomeToPremiumProps) => {
+  const l10n = useL10n();
+  const sumOfReduction = props.data.reduce(
+    (total, [_label, num]) => total + num,
+    0
+  );
+  const percentages = props.data.map(([label, num]) => {
+    return [label, num / sumOfReduction] as const;
+  });
+
+  const diameter = 100;
+  const ringWidth = 15;
+  const radius = (diameter - ringWidth) / 2;
+  const circumference = 2 * Math.PI * radius;
+  const sliceBorderWidth = 0;
+  const headingNumberSize = diameter / 8;
+  const headingLabelSize = headingNumberSize / 2;
+  const headingGap = 4;
+
+  const slices = percentages.map(([label, percent], index) => {
+    const percentOffset = percentages
+      .slice(0, index)
+      .reduce((offset, [_label, num]) => offset + num, 0);
+
+    return (
+      <circle
+        key={label}
+        cx={diameter / 2}
+        cy={diameter / 2}
+        r={radius}
+        className={styles.slice}
+        fill="none"
+        strokeWidth={ringWidth}
+        strokeDasharray={`${circumference} ${circumference}`}
+        style={
+          {
+            "--sliceLength": circumference * (1 - percent) + sliceBorderWidth,
+          } as CSSProperties
+        }
+        // Rotate it to not overlap the other slices
+        transform={`rotate(${-90 + 360 * percentOffset} ${diameter / 2} ${
+          diameter / 2
+        })`}
+      />
+    );
+  });
+
+  return (
+    <>
```

Comment:
Q: Do we need the wrapper `<>` here, or can we just return the `<figure>` directly since it's the only parent?

Reply:
Fixed thanks